nginx-controller/controller/controller.go

```diff
@@ -661,50 +722,93 @@ func (lbc *LoadBalancerController) syncSecret(task Task) {
 			if err := lbc.cnf.AddOrUpdateSecret(secret); err != nil {
 				glog.Errorf("Error when updating Secret %v: %v", key, err)
 				lbc.recorder.Eventf(secret, api_v1.EventTypeWarning, "UpdatedWithError", "%v was updated, but not applied: %v", key, err)
-				for _, ing := range ings {
+				for _, ing := range nonMinions {
 					lbc.recorder.Eventf(&ing, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for %v/%v was updated, but not applied: %v", ing.Namespace, ing.Name, err)
 				}
+				for _, minion := range minions {
+					master, err := lbc.findMasterForMinion(&minion)
+					if err != nil {
+						glog.Errorf("Ignoring Ingress %v(Minion): %v", minion.Name, err)
+						continue
+					}
+					lbc.recorder.Eventf(master, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for  minion %v/%v was updated, but not applied: %v", minion.Namespace, minion.Name, err)
+					lbc.recorder.Eventf(&minion, api_v1.EventTypeWarning, "UpdatedWithError", "Configuration for %v/%v was updated, but not applied: %v", minion.Namespace, minion.Name, err)
+				}
 			} else {
 				lbc.recorder.Eventf(secret, api_v1.EventTypeNormal, "Updated", "%v was updated", key)
-				for _, ing := range ings {
+				for _, ing := range nonMinions {
 					lbc.recorder.Eventf(&ing, api_v1.EventTypeNormal, "Updated", "Configuration for %v/%v was updated", ing.Namespace, ing.Name)
 				}
+				for _, minion := range minions {
+					master, err := lbc.findMasterForMinion(&minion)
+					if err != nil {
+						glog.Errorf("Ignoring Ingress %v(Minion): %v", minion.Name, err)
+						continue
+					}
+					lbc.recorder.Eventf(master, api_v1.EventTypeNormal, "Updated", "Configuration for minion %v/%v was updated", minion.Namespace, minion.Name)
+					lbc.recorder.Eventf(&minion, api_v1.EventTypeNormal, "Updated", "Configuration for %v/%v was updated", minion.Namespace, minion.Name)
+				}
 			}
 		}
 	}
 }
 
-func (lbc *LoadBalancerController) findIngressesForSecret(secret string) ([]extensions.Ingress, error) {
-	res := []extensions.Ingress{}
+func (lbc *LoadBalancerController) findIngressesForSecret(secret string) ([]extensions.Ingress, []extensions.Ingress, error) {
```

Comment:
I think we should use named return types so that nonMinions and minions are not mixed up in future